### PR TITLE
93074 Update travel claims client to take in client_number as initialization parameter

### PR DESCRIPTION
**Pull request by randomsync**
## Summary
We need to use a different BTSSS API client_number when filing travel claims for OH sites, as compared to VistA sites. This PR adds an initialization parameter to the client class, which the service class can use to initialize the client with a different client_number.

This is a backward compatible change and doesn't require a feature flag.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/93074

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
* check_in experience and Oracle Health travel claims

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).

### CODE REVIEW

**Review by ndouglas — DISMISSED** [review]

File: modules/check_in/app/services/travel_claim/client.rb [hunk 1]
```diff
@@ -12,9 +12,9 @@ class Client
     CLAIMANT_ID_TYPE = 'icn'
     TRIP_TYPE = 'RoundTrip'
 
-    attr_reader :settings, :check_in
+    attr_reader :settings, :check_in, :client_id
```
Comment (kanchanasuriya) [inline modules/check_in/app/services/travel_claim/client.rb:15]: Ticket says "We need to use a different client number for submitting claims for Oracle Health sites."; 

Could you please clarify if we need to use a different `client_id` or `client_number` for oracle healthcare?

Comment (randomsync) [reply modules/check_in/app/services/travel_claim/client.rb]: You're right, I confused `client_id` with `client_number`. `client_number` is what we need to parameterize. corrected in a62248e51d9616f707877e368678eb4cde17b915

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Review by LindseySaari — APPROVED** [review]

**Comment by randomsync** [comment]
@LindseySaari thank you for approving. the backend-review-group approval CI check is still failing and blocking the merge of this. is there anyone else who needs to approve?

**Pull request merged**